server/handlers/validation.go

```diff
@@ -80,9 +80,21 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 			return nil, ErrBadTargets{msg: err.Error()}
 		}
 		repo.SetTargets(targetsRole, t)
+	} else {
+		targetsJSON, err := store.GetCurrent(gun, data.CanonicalTargetsRole)
+		if err != nil {
+			return nil, fmt.Errorf("could not load repo for snapshot generation: %v", err)
```

Comment:
Shouldn't it be a validation error if we can't load the targets, partially because then we can't validate the snapshot, but also because we need to also validate the old target file.  (We should probably load and validate all 4 pieces of metadata as a whole - for instance, I think currently, if we rotate the targets key and submit a new root.json, but don't submit a targets.json, which seems wrong because then the repo will be in a broken state?)